plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java

```diff
@@ -182,6 +202,25 @@ public String getRegistryURL() {
 			return "";
 		}
 		
+        public void setClusterNamespace(String value) {
+            Connection connection = getConnection();
+            if(connection != null) {
+                Map<String, Object> properties = connection.getExtendedProperties();
+                Object old = properties.get(ICommonAttributes.CLUSTER_NAMESPACE_KEY);
+                connection.setExtendedProperty(ICommonAttributes.CLUSTER_NAMESPACE_KEY, value);
+                firePropertyChange(PROP_CLUSTER_NAMESPACE, old, value);
+            }           
+        }
+        
+        public String getClusterNamespace() {
+            Connection connection = getConnection();
+            if(connection != null) {
+                return connection.getClusterNamespace(); 
+            }
+            return "";
```

Comment:
You are theorically right although I practically that it will never happen because we are editing a connection but will add for code safety